### PR TITLE
Bug 1877441: Watch for changes to Tasks & ClusterTasks when in pipeline builder

### PR DESCRIPTION
**Pull request by rottencandy**
**Fixes**: 
https://issues.redhat.com/browse/ODC-3171

**Analysis / Root cause**: 
List of Tasks and ClusterTasks is only fetched once when the pipeline builder page is loaded.

**Solution Description**: 
Watch for changes in Tasks and ClusterTasks in the pipeline builder.

**Screen shots / Gifs for design review**: 
demo

**Unit test coverage report**: 
Unchanged

**Browser conformance**: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts [hunk 1]
```diff
@@ -40,62 +41,32 @@ type UseTasks = {
...
+        namespaced: false,
+      },
+    }),
+    [namespace],
+  );
+  const { tasks, clusterTasks } = useK8sWatchResources<{ [kind: string]: PipelineResourceTask[] }>(
+    memoizedResources,
+  );
+  let errorMsg;
+  if (tasks.loaded && tasks.loadError) {
```
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts:59]:
```suggestion
  if (tasks.loadError) {
```

File: frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts [hunk 2]
```diff
@@ -40,62 +41,32 @@ type UseTasks = {
...
+    errorMsg = `Failed to load namespace Tasks. ${tasks.loadError}`;
+  }
+  if (clusterTasks.loaded && clusterTasks.loadError) {
```
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts:62]:
```suggestion
  if (clusterTasks.loadError) {
```

File: frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts [hunk 3]
```diff
@@ -40,62 +41,32 @@ type UseTasks = {
...
+    errorMsg = `Failed to load ClusterTasks. ${clusterTasks.loadError}`;
+  }
 
   return {
-    namespacedTasks,
-    clusterTasks,
-    errorMsg: loadErrorMsg,
+    namespacedTasks: tasks.loaded ? tasks.data : null,
+    clusterTasks: clusterTasks.loaded ? clusterTasks.data : null,
```
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts:68]:
```suggestion
    namespacedTasks: tasks.loaded && !tasks.loadError ? tasks.data : null,
    clusterTasks: clusterTasks.loaded && !clusterTasks.loadError ? clusterTasks.data : null,
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Comment by debsmita1** [comment]
this works fine
/lgtm

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**